EarlGrey/Additions/UIViewController+GREYAdditions.m

```diff
@@ -179,14 +183,18 @@ - (void)greyswizzled_viewWillDisappear:(BOOL)animated {
     // Interactive transitions can cancel and cause imbalance of will and did calls.
     id<UIViewControllerTransitionCoordinator> coordinator = [self transitionCoordinator];
     if (coordinator && [coordinator initiallyInteractive]) {
-      [coordinator notifyWhenInteractionEndsUsingBlock:
-          ^(id<UIViewControllerTransitionCoordinatorContext> context) {
-            if ([context isCancelled]) {
-              GREYAppStateTrackerObject *object =
-                  objc_getAssociatedObject(self, @selector(greyswizzled_viewWillAppear:));
-              UNTRACK_STATE_FOR_OBJECT(kGREYPendingViewsToDisappear, object);
-            }
-          }];
+      id block = ^(id<UIViewControllerTransitionCoordinatorContext> context) {
+        if ([context isCancelled]) {
+          GREYAppStateTrackerObject *object =
+              objc_getAssociatedObject(self, @selector(greyswizzled_viewWillAppear:));
+          UNTRACK_STATE_FOR_OBJECT(kGREYPendingViewsToDisappear, object);
+        }
+      };
+#if !defined(__IPHONE_10_0) || __IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_10_0
+      [coordinator notifyWhenInteractionEndsUsingBlock:block];
+#else
+      [coordinator notifyWhenInteractionChangesUsingBlock:block];
+#endif
```

Comment:
Sorry for the late comment, but this could also be written as:

      if @available(iOS 10, *) {
        [coordinator notifyWhenInteractionChangesUsingBlock:block];
      }
    #if !defined(__IPHONE_10_0) || __IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_10_0
      else {
        [coordinator notifyWhenInteractionEndsUsingBlock:block];
      }
    #endif

This would have the advantage of invoking the iOS10 API on ios 10 and above even if the code is compiled for a minimum target or 8 or 9.

Reply:
@e-noyau  -
I went with the current implementation primarily to make the code a little more readable (my opinion). This also will maintain existing behavior for current iOS 10 users who are using 8.0 as the deployment target.

I understand the advantage of your suggestion, I'll let @tirodkar or @sskhandp decide which they prefer. I'm happy to change it.

Thanks!

Reply:
TBH - either is fine with me but we haven't use availability macros like @e-noyau anywhere in the codebase before so slightly leaning towards @mbaxley's solution.